session.go

```diff
@@ -155,7 +220,7 @@ func (s *Session) NewSender(ctx context.Context, target string, opts *SenderOpti
 	if err != nil {
 		return nil, err
 	}
-	if err = l.attach(ctx, s); err != nil {
+	if err = l.attach(ctx); err != nil {
```

Comment:
'l' -> 's'

Reply:
I intend to clean these up in a separate pass, else the diff becomes way too noisy.

Reply:
Yeah, it's minor, but it's also a reward for the work you did ;)